### ISSUE TITLE
Postgres dialect: sqlfluff erroneously reports "execution" as a keyword triggering rule L029

### ISSUE THREAD

**Issue by frederikaalund** [sqlfluff/sqlfluff#4269]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

sqlfluff erroneously reports "execution" as a keyword for the postgres dialect.

### Expected Behaviour

No reported errors.

Note that `execution` is *not* a keyword in postgres: https://www.postgresql.org/docs/current/sql-keywords-appendix.html
I know that both `exec` and `execute` *are* keywords but not `execution`.

### Observed Behaviour

I get L029 "Keywords should not be used as identifiers.".

### How to reproduce

Try with this SQL file:

```sql
CREATE TABLE IF NOT EXISTS execution (
    execution_id UUID PRIMARY KEY
);
```

and use the setting `unquoted_identifiers_policy = "all"`

### Dialect

postgres

### Version

1.4.5

### Configuration

Snippet of my `pyproject.toml` file (enough to reproduce the issue):
```
[tool.sqlfluff.core]
dialect = "postgres"

[tool.sqlfluff.rules.L029]
# Keywords should not be used as identifiers regardless of whether we
# quote said identifiers or not.
unquoted_identifiers_policy = "all"
```

Note that there is no problem if I don't change `unquoted_identifiers_policy`.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
